pkg/util/querylimits/propagation.go

```diff
@@ -21,11 +21,12 @@ const (
 // NOTE: we use custom `model.Duration` instead of standard `time.Duration` because,
 // to support user-friendly duration format (e.g: "1h30m45s") in JSON value.
 type QueryLimits struct {
-	MaxQueryLength          model.Duration `json:"maxQueryLength,omitempty"`
-	MaxQueryLookback        model.Duration `json:"maxQueryLookback,omitempty"`
-	MaxEntriesLimitPerQuery int            `json:"maxEntriesLimitPerQuery,omitempty"`
-	QueryTimeout            model.Duration `json:"queryTimeout,omitempty"`
-	RequiredLabels          []string       `json:"requiredLabels,omitempty"`
+	MaxQueryLength          model.Duration   `json:"maxQueryLength,omitempty"`
+	MaxQueryLookback        model.Duration   `json:"maxQueryLookback,omitempty"`
+	MaxEntriesLimitPerQuery int              `json:"maxEntriesLimitPerQuery,omitempty"`
+	QueryTimeout            model.Duration   `json:"queryTimeout,omitempty"`
+	RequiredLabels          []string         `json:"requiredLabels,omitempty"`
+	RequiredNumberLabels    int              `json:"requiredNumberLabelsMatchers,omitempty"`
```

Comment:
one more: the validation limits expects `...label_matchers` instead of `labels_matchers`
```suggestion
	RequiredNumberLabels    int              `json:"requiredNumberLabelMatchers,omitempty"`
```